pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -124,11 +124,14 @@ func getSubnet(controlPlaneSubnets []*subnets.Subnet, subnetID, subnetName strin
 }
 
 // validateExternalNetwork validates the user's input for the externalNetwork and returns a list of all validation errors
-func validateExternalNetwork(p *openstack.Platform, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
+func validateExternalNetwork(p *openstack.Platform, ci *CloudInfo, n *types.Networking, fldPath *field.Path) (allErrs field.ErrorList) {
 	// Return an error if external network was specified in the install config, but hasn't been found
 	if p.ExternalNetwork != "" && ci.ExternalNetwork == nil {
 		allErrs = append(allErrs, field.NotFound(fldPath.Child("externalNetwork"), p.ExternalNetwork))
 	}
+	if p.ExternalNetwork != "" && len(n.MachineNetwork) == 1 && n.MachineNetwork[0].CIDR.IPNet.IP.To4() == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("externalNetwork"), p.ExternalNetwork, "External network not necessary on Single Stack IPv6 cluster"))
```

Comment:
since it's an error, I'll find a stronger word than "necessary". In this case, it's actually "Invalid".